site/docs/references/pseudo-classes.md

```diff
@@ -38,12 +38,248 @@ To define custom pseudo-classes, or states, for a simple selector, you tell **St
 .Example1__root[data-Example1-loading][data-Example1-toggled] { color: blue; }
 ```
 
+
+
+## Custom pseudo-classes with parameters
+
+in some use cases its useful to define custom states that use a parameter to indicate which nodes to activate on. 
+
+for example a cell in a grid can be marked using column and row pseudo-classes
+
+```css
+/* stateWithNumberParam.st.css */
+.cell{
+    -st-states: column(number), 
+                row(number) 
+}
+
+.cell:column(1):row(1){
+    color:red;
+}
+
+```
+
+```css
+/* CSS output*/
+.Example1__root[data-Example1-column="1"][data-Example1-row="1"] { color: red; }
+
+```
+
+### Types and allowed prefixes
+
+***Stylable*** supports a number of parameter types for pseudo-classes, these types allow us to provide better tooling when using the pseudo-class.
+
+in the future we intend to support extra functunality for the numeral types through build time tricks
+
+
+| Type | Allowed validations | Allowed prefixes |
+|----|----|----|
+| string | minLength <br> maxLength | "~" - match whole words <br> "^" - match start <br> "$" - match end <br> "*" - match include |
+| number | minimum <br> maximum <br> multipleOf <br> | \> - greater then (future) <br> \< - lesser then (future) <br> n+1 - [nth child format](https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child) (future)  |
+| boolean | - | - |
+| tag  | - | - |
+| enum | options | - |
+| percentage | - | \> - greater then - (future) <br> \< - lesser then - (future) |
+
+
+
+
+
+#### String example
+
+```css
+/* defining state */
+.token{
+    -st-states: fieldName(string);
+}
+
+/* customize fields with fieldName email */
+.token:fieldName(email){
+    color:lightBlue;
+}
+
```

Comment:
Make string glob syntax identical to CSS.
fieldname($_id) does not look like it means 'ends with _id'
Make it fieldname($=_id) like CSS.

Reply:
I agree that looks better, and more consistent with CSS.